### PR TITLE
More bugfixes for pilot with spaceship

### PR DESCRIPTION
**Pull request by KrauseFx**

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Pull Request is Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a changelog summary in the Pull Request description @KrauseFx</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: spaceship/lib/spaceship/test_flight/group.rb [hunk 1]
```diff
@@ -37,11 +37,11 @@ def self.filter_groups(app_id: nil, &block)
     end
 
     def add_tester!(tester)
-      client.add_tester_to_group!(group: self, tester: tester, app_id: self.app_id)
+      client.put_test_to_group(group_id: self.id, tester_id: tester.tester_id, app_id: self.app_id)
     end
 
     def remove_tester!(tester)
-      client.remove_tester_from_group!(group: self, tester: tester, app_id: self.app_id)
+      client.delete_tester_from_group(group_id: self.id, tester_id: tester.tester_id, app_id: self.app_id)
```
Comment (hjanuschka) [inline spaceship/lib/spaceship/test_flight/group.rb]: wow, we should have seen that 🤒

Comment (KrauseFx) [inline spaceship/lib/spaceship/test_flight/group.rb]: Yeah, the methods were renamed in client.rb, and we haven't updated the parts that call it...

File: spaceship/lib/spaceship/test_flight/group.rb [hunk 2]
```diff
@@ -36,12 +36,20 @@ def self.filter_groups(app_id: nil, &block)
...
 
+    # First we need to add the tester to the app
+    # It's ok if the tester already exists, we just have to do this... don't ask
+    # This will enable testing for the tester for a given app, as just creating the tester on an account-level
+    # is not enough to add the tester to a group. If this isn't done the next request would fail.
+    # This is a bug we reported to the iTunes Connect team, as it also happens on the iTunes Connect UI on 18. April 2017
     def add_tester!(tester)
-      client.add_tester_to_group!(group: self, tester: tester, app_id: self.app_id)
+      # This post request makes the account-level tester available to the app
+      client.post_tester(app_id: self.app_id, tester: tester)
```
Comment (KrauseFx) [inline spaceship/lib/spaceship/test_flight/group.rb]: Nice 👍

File: spaceship/lib/spaceship/tunes/tester.rb [hunk 3]
```diff
@@ -106,8 +106,9 @@ def create!(email: nil, first_name: nil, last_name: nil, groups: nil)
                             first_name: first_name,
                              last_name: last_name,
                                 groups: groups)
-          # We don't need to do additional parsing here, as `create_tester!` automatically
-          # re-fetches the newly created tester
+          # The response of the client request is a hash but doesn't contain all of the necessary data (like testerId, etc)
+          # use #find to get an instance of Tester with all of the data.
+          find(email)
```
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tester.rb]: Cool 👍

**Review by KrauseFx — COMMENTED** [review]
Great work

**Review by ohayon — APPROVED** [review]

**Pull request merged**